test/e2e/data/eks/cluster-template-eks-managed-machinepool-with-launch-template-only.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachinePool
+metadata:
+  name: "${CLUSTER_NAME}-pool-0"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: ${WORKER_MACHINE_COUNT}
+  template:
+    spec:
+      version: "EKS_KUBERNETES_VERSION"
+      clusterName: "${CLUSTER_NAME}"
+      bootstrap:
+        dataSecretName: "${CLUSTER_NAME}-pool-0-userdata"
```

Comment:
This would not work here. The existing behavior requires the user to specify a data secret used as bootstrap data in AWSMachinePool (https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/scope/machinepool.go#L126)